src/api-handler.ts

```diff
@@ -127,34 +121,37 @@ export class ApiHandler implements MagicalClass {
      *
      * @param {*} args      Arguments
      * @returns {Promise}   Resolvable API provider promise
-     * @memberof ApiHandler
      */
     public async handleRequest(...args: any): Promise<IRequestResponse> {
         const prop = args[0];
-        const api = this.apiEndpoints[prop];
+        const endpointSettings = this.apiEndpoints[prop];
 
         const queryParams = args[1] || {};
         const uriParams = args[2] || {};
         const requestConfig = args[3] || {};
 
-        const uri = api.url.replace(/:[a-z]+/ig, (str: string) => (uriParams[str.substr(1)] ? uriParams[str.substr(1)] : str));
-
-        const requestData = {
-            ...queryParams,
-        };
+        const uri = endpointSettings.url.replace(/:[a-z]+/ig, (str: string) => (uriParams[str.substr(1)] ? uriParams[str.substr(1)] : str));
 
         let responseData = null;
 
-        responseData = await this.httpRequestHandler[api.method](uri, requestData, requestConfig);
+        const additionalRequestSettings = { ...endpointSettings };
+
+        delete additionalRequestSettings.url;
+        delete additionalRequestSettings.method;
+
+        responseData = await this.httpRequestHandler[endpointSettings.method](uri, queryParams, {
+            ...requestConfig,
+            ...additionalRequestSettings,
```

Comment:
I would go with `Object.assign` in this case rather then this, `.assign` sometimes is better as it does clone both entries and merge into new object.

`With Object.assign(), you would have to ensure you always pass an empty object {} as the first argument`, ensuring in a lib always better

Reply:
They're interchangeable in fact. We get cloning from both. I think that this way is just more POJO. We can think about avoiding cloning maybe. Let me know if you have some ideas pls

Reply:
Perhaps they're more POJO, still I think the traditional way of `Object.assign` could be better to ensure for being assigned to empty object. Yeah, we could think of some ways of preventing cloning and having additional reference address over this as well in this regards.